sharkfuser/benchmarks/driver.cpp

```diff
@@ -315,9 +575,24 @@ int main(int argc, char **argv) {
       // When unspecified, default to fp32 conv.
       convIOType = DataType::Float;
 
-    auto status = benchmarkConvFprop(n, c, d, h, w, g, k, z, y, x, t, u, v, o,
-                                     p, q, m, l, j, imageLayout, outputLayout,
-                                     filterLayout, s, bias, iter, convIOType);
+    ErrorObject status = ok();
+    if (mode == 1) {
+      // Forward convolution
+      status = benchmarkConvFprop(n, c, d, h, w, g, k, z, y, x, t, u, v, o, p,
+                                  q, m, l, j, imageLayout, outputLayout,
+                                  filterLayout, s, bias, iter, convIOType);
+    } else if (mode == 2) {
+      // Weight gradient
+      status = benchmarkConvWGrad(n, c, d, h, w, g, k, z, y, x, t, u, v, o, p,
+                                  q, m, l, j, imageLayout, outputLayout,
+                                  filterLayout, s, iter, convIOType);
+    } else if (mode == 4) {
+      // Data gradient
+      status = benchmarkConvDGrad(n, c, d, h, w, g, k, z, y, x, t, u, v, o, p,
+                                  q, m, l, j, imageLayout, outputLayout,
+                                  filterLayout, s, iter, convIOType);
+    }
```

Comment:
From the BOO driver / MIOpen [options](https://github.com/ROCm/rocm-libraries/blob/4a426fb4aa53aab7b91cdf154875dbdda54e3c39/projects/miopen/driver/conv_driver.hpp#L908), this appears to be flipped. i.e. they use `F=2` for BWD (ConvDGrad) and `F=4` for WRW (ConvWGrad).